Girder roles

opadron opened the pull request:
Eighth among a series of PR's.

Adds a new role: `girder`, for provisioning and managing girder nodes.

Initially contains the commits from all prior PR's in the series (will be rebased after they are merged).

New commits: 1

Commit:
add girder roles

opadron commented:
rebased after merging #21

Commit:
change girder_group default to the correct value in the girder documentation

Commit:
add proper documentation for the girder_replica_set option in the girder role

zachmullen commented on roles/girder-install/tasks/main.yml:
```diff
@@ -0,0 +1,228 @@
+      - mongodb
+      - ncurses-dev
+      - openjdk-7-jdk
+      - openjdk-7-jre-headless
```
Why do we need gfortran, autoconf, sqlite, libexpat, mongodb, ncurses-dev, and openjdk?

opadron replied:
I don't think we do.  I just haven't been keeping close tabs on the dependencies.  Thanks for pointing it out.

opadron commented:
I've added #28 to handle the issue of package dependencies in a more comprehensive manner.  Considering that the dependency issue spans multiple roles, many of which have already been merged from prior PRs, I propose we punt on the dependency issue and move the discussion to the new issue.

zachmullen commented on roles/girder-post-install/files/girder-post-install.py:
```diff
@@ -0,0 +1,168 @@
+    return rest(client, 'put', *args, **kwds)
+
+def get(client, *args, **kwds):
+    return rest(client, 'get', *args, **kwds)
```
We actually have all these methods inside `GirderClient` already. https://github.com/girder/girder/blob/master/clients/python/girder_client/__init__.py#L228-L238

Commit:
rewrite girder-post-install script to remove superfluous functions

kotfic commented:
@opadron  @cpatrick I've been able to successfully deploy this role on a vagrant box and verify that everything is working. My concerns about this role fall into three categories:
1. inclusion of romanesco, climos_test and spark_test plugins which are not relevant to a general base installation of girder
2. Reliance on HDFS and HDFS asset stores to successfully run this role
3. Use of the `girder-post-install.py` file to configure the girder install

As I understand from @opadron  1 & 2 are needed for gobig and are known problems.  I would suggest adding refactoring issues about plugin and HDFS dependencies and then explicitly mention these in the girder documentation before the table of variables.  (E.g., _Note: This role depends on [plugins] and presence of a working HDFS installation -  please see issues [list issues]_).

My concern with 3 is the use of a non-trival python script to configure girder-  which should be the exclusive responsibility of the deployment mechanism (ansible). With that said,  reviewing covalic and HPCCloud deployments of girder+plugins it doesn't appear that there is a better solution than the ``girder-post-install` script.  I have started work on my own time to implement an ansible module wrapper around girder client that might resolve some of these issues.  In any case I don't think 3 should  block merging this branch.

If you can add the issues and a note in the documentation than this LGTM.

Commit:
add notice to girder role documentation

opadron commented:
@kotfic we are totally in agreement.  I have added a new issue and updated the documentation for the girder role.  If you can make a final pass through these two items and are satisfied, I will pull the trigger on this PR and move on to our growing laundry list. :)

kotfic commented:
LGTM
